src/priceHistory/priceHistoryChart.tsx

```diff
@@ -0,0 +1,356 @@
+            <Line x1={0} y1={y} x2={chartWidth} y2={y} stroke={colors.gray2} strokeWidth="1" />
+            <Circle cx={x} cy={y} r="4" fill={color} />
+          </G>
+        )
+        break
+
+      case chartData.length - 1:
+        result.push(
+          <G key={idx + 'dot'}>
+            <Circle cx={x} cy={y} r="4" fill={color} />
+          </G>
+        )
+        break
+    }
+    switch (idx) {
+      case highestRateIdx:
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[highestRateIdx].displayValue}
+            position={'top'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+
+      case lowestRateIdx:
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[lowestRateIdx].displayValue}
+            position={'bottom'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+    }
+
+    switch (result.length) {
+      case 0:
+        return null
+      case 1:
+        return result[0]
+      default:
+        return <>{result}</>
+    }
+  }
+}
+
+interface PriceHistoryChartProps {
+  tokenId: string
+  containerStyle?: ViewStyle
+  testID?: string
+  chartPadding?: number
+  color?: colors
+}
+
+export default function PriceHistoryChart({
+  tokenId,
+  containerStyle,
+  testID,
+  chartPadding,
+  color = colors.black,
+}: PriceHistoryChartProps) {
+  const priceHistoryPrices = useSelector((state: RootState) =>
+    priceHistoryPricesSelector(state, tokenId)
+  )
+
+  const priceHistoryStatus = useSelector((state: RootState) =>
+    priceHistoryStatusSelector(state, tokenId)
+  )
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const localExchangeRate = useSelector(usdToLocalCurrencyRateSelector)
+  const dispatch = useDispatch()
+
+  const dollarsToLocal = useCallback(
+    (amount: BigNumber.Value | null) =>
+      convertDollarsToLocalAmount(amount, localCurrencyCode ? localExchangeRate : 1),
+    [localExchangeRate]
+  )
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
+
+  useEffect(() => {
+    const latestTimestamp = priceHistoryPrices.at(-1)?.priceFetchedAt ?? 0
+    if (priceHistoryPrices.length > 0 && latestTimestamp > Date.now() - ONE_HOUR_IN_MILLIS) {
+      return
+    }
+
+    dispatch(
+      fetchPriceHistoryStart({
+        tokenId,
+        startTimestamp: Date.now() - ONE_DAY_IN_MILLIS * 30,
+        endTimestamp: Date.now(),
+      })
+    )
+  }, [])
+
+  if (priceHistoryStatus === 'loading' && priceHistoryPrices.length === 0) {
+    return <Loader />
+  } else if (priceHistoryPrices.length === 0) {
+    return null
+  }
+
+  // Create chart data from price history
+  const chartData = []
```

Comment:
This section creating the chartData is largely new instead of relying on the `aggregateExchangeRates` present in `src/exchange/reducer.ts`. This is preferable as if we ever have data issue we can populate the fixed the data from the API and if the user is unable to fetch new prices we will use the prices already present in redux.

Reply:
this seems a bit dense, should this be in its own function with unit tests?